### DISCUSSION TITLE
Does xrust support namespace?

### DISCUSSION THREAD

**Discussion by tacoberu** [ballsteve/xrust#57] [Q&A]
Hello.

I would like to use xrust in my library for xslt transformations. But somehow I can't force namespaces to be used on the output. Wouldn't there be a sample somewhere?

Thanks for the direction.

**Comment by Devasta** [comment]
Hi @tacoberu,

xrust is still a work in progress so there is a lot that is still up in the air, but it should be outputting namespaces; would it be possible to get a sample of what you are running?

**Comment by ballsteve** [comment]
As Daniel says, this should be working. Xrust may be incomplete, but the features that are present should work as per the W3C specs.

Wherever this is not the case, then please raise an issue. A bug report issue should include, at the minimum, the source document, XSL stylesheet, and the expected result document.

**Comment by ballsteve** [comment]
I've created [issue 58](https://github.com/ballsteve/xrust/issues/58) for this bug.

Please add source document, XSL stylesheet, and expected result document. We will then add this as a test in the crate.

I'm working on the next release of the crate, and will make sure this issue is fixed.

**Comment by tacoberu** [comment]
Thank you very much for your response. I wasn't sure how to react. If it was known that it was not supposed to work, it would be pointless to file a bug report. So apparently I'm using it wrong and missed something. So I will continue to issue 58.
Thanks again. And thanks to your work. I really like xrust.

**Reply by ballsteve** [reply]
Don't hesitate to open an issue or report a bug. I'm only too happy to close an issue, with an explanation as to why it is being closed.

The worst thing would be not know in the first place that someone is having a problem.

If something is known to not work, then that should be in the documentation. If it is not then you would open a 'documentation' bug report.
